classes/class-wc-connect-cart-validation.php

```diff
@@ -11,13 +11,6 @@ class WC_Connect_Cart_Validation {
 	 */
 	private $current_package;
 
-	/**
-	 * Register actions when required instead of using constructor.
-	 */
-	public function register_actions() {
-		add_action( 'woocommerce_store_api_cart_errors', [ $this, 'add_api_cart_errors' ], 10, 2 );
```

Comment:
do you have any idea why its added in the first place ? do you think we should check this [commit](https://github.com/Automattic/woocommerce-services/commit/47ab8331da19b100f9a8b01e37546b9b9ad2f697)

Reply:
It was added back in the day when WooCommerce still need to use WooCommerce Blocks plugin to display the error. But upon my testing on the latest WooCommerce this action is seems not needed anymore. It can check the error by itself.

That's why I'm proposing to remove this action. But after having a second thought. I'll add this action again and add some condition to make this plugin compatible with older WooCommerce and WooCommerce Blocks plugin.

Reply:
it looks good, I tested it and the issue fixed